upgrade-2.x.sh

```diff
@@ -156,9 +156,9 @@ function compare_device_state() {
 function stop_services() {
     # Stopping supervisor and related services
     log "Stopping supervisor and related services..."
-    systemctl stop update-resin-supervisor.timer > /dev/null 2>&1
-    systemctl stop resin-supervisor > /dev/null 2>&1
-    ${DOCKER_CMD} rm -f resin_supervisor > /dev/null 2>&1 || true
+    systemctl stop update-balena-supervisor.timer > /dev/null 2>&1 || systemctl stop update-resin-supervisor.timer > /dev/null 2>&1
+    systemctl stop balena-supervisor  > /dev/null 2>&1 || systemctl stop resin-supervisor > /dev/null 2>&1
```

Comment:
`errexit` is set so we can't allow an exit code 1 in here.

- `systemctl stop balena-supervisor resin-supervisor` will exit 1
- `systemctl stop balena-supervisor resin-supervisor || true` won't catch failures to stop the service